### ISSUE TITLE
FalsePositive for UseDataClass on enum and annotation classes

### ISSUE THREAD

**Issue by skimarxall** [arturbosch/detekt#435]
Maybe I am missing some concept but I am getting warning for enum class that could be converted into data class.

I don't know how.

```kotlin
enum class Environment(val appId: String,
                       val appCode: String,
                       val aUrl: String,
                       val bUrl: String,
                       val cUrl: String
) {

    PRODUCTION("", "", "", "", ""),
    STAGING("","","","","" );
}
```

**Comment by schalkms** [comment]
Thanks for reporting!
I think enums should be excluded from this rule.

**Comment by skimarxall** [comment]
Annotation class should be added to the list too.
Shall I create new issue?

```kotlin
@MustBeDocumented
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
@MapKey
internal annotation class ViewModelKey(val value: KClass<out ViewModel>)
```

**Comment by schalkms** [comment]
I think this belongs to #377

**Comment by arturbosch** [comment]
Not quite. Annotation class != annotated class. This issue will be enough, I change the name. Thanks for reporting!

**Comment by schalkms** [comment]
Oh yes. I read over the `annotation`.

**Comment by skimarxall** [comment]
Thanks for the fix
